soilmask.py

```diff
@@ -0,0 +1,206 @@
+import pathlib
+
+import numpy as np
+import logging
+import cv2
+import tempfile
+from osgeo import gdal, osr
+
+MAX_PIXEL_VAL = 255
+
+
+def gen_rgb_mask(img: np.ndarray, bin_mask: np.ndarray) -> np.ndarray:
+    """Applies the mask to the image
+    Arguments:
+        img: the source image to mask
+        bin_mask: the mask to apply to the image
+    Return:
+        A new image that had the mask applied
+    """
+    rgb_mask = cv2.bitwise_and(img, img, mask=bin_mask)
+
+    return rgb_mask
+
+
+def gen_plant_mask(color_img: np.ndarray, kernel_size: int = 3) -> np.ndarray:
+    """Generates an image with plants masked in.
+    Arguments:
+        color_img: RGB image to mask
+        kernel_size: masking kernel size
+    Return:
+        An RGB image with plants masked in
+    """
+    r_channel = color_img[:, :, 2]
+    g_channel = color_img[:, :, 1]
+    b_channel = color_img[:, :, 0]
+
+    sub_img = (g_channel.astype('int') - r_channel.astype(
+        'int') - 0) > 0  # normal: -2
+
+    mask = np.zeros_like(b_channel)
+
+    mask[sub_img] = MAX_PIXEL_VAL
+
+    blur = cv2.blur(mask, (kernel_size, kernel_size))
+    pix = np.array(blur)
+    sub_mask = pix > 128
+
+    mask_1 = np.zeros_like(b_channel)
+    mask_1[sub_mask] = MAX_PIXEL_VAL
+
+    return mask_1
+
+def get_epsg(filename):
```

Comment:
Would it be possible to add Python 3 type annotations like the other functions have?